runtime/vm/jfr.cpp

```diff
@@ -544,6 +545,40 @@ jfrVMInitialized(J9HookInterface **hook, UDATA eventNum, void *eventData, void *
 	}
 }
 
+/**
+ * Hook for VM monitor waited. Called without VM access.
+ *
+ * @param hook[in] the VM hook interface
+ * @param eventNum[in] the event number
+ * @param eventData[in] the event data
+ * @param userData[in] the registered user data
+ */
+static void
+jfrVMMonitorWaited(J9HookInterface **hook, UDATA eventNum, void *eventData, void *userData)
+{
+	J9VMMonitorWaitedEvent *event = (J9VMMonitorWaitedEvent*) eventData;
+	J9VMThread *currentThread = event->currentThread;
+	PORT_ACCESS_FROM_VMC(currentThread);
+
+#if defined(DEBUG)
+	j9tty_printf(PORTLIB, "\n!!! VM monitor waited %p\n", currentThread);
+#endif /* defined(DEBUG) */
+
+	J9JFRMonitorWaited *jfrEvent = (J9JFRMonitorWaited*)reserveBufferWithStackTrace(currentThread, currentThread, J9JFR_EVENT_TYPE_OBJECT_WAIT, sizeof(*jfrEvent));
+	if (NULL != jfrEvent) {
+		jfrEvent->time = (event->millis * 1000000) + event->nanos;
+		jfrEvent->duration = 0;
+		jfrEvent->duration = j9time_nano_time() - event->startTicks;
+		jfrEvent->monitorClass = event->monitorClass;
+		jfrEvent->monitorAddress = event->monitorAddress;
+
+		if (J9THREAD_TIMED_OUT == event->reason) {
```

Comment:
The buffer entry is not zeroed, so you should probably just assign this unconditionally:

```
jfrEvent->timedOut = (J9THREAD_TIMED_OUT == event->reason);
```